aten/src/ATen/core/jit_type.h

```diff
@@ -1296,13 +1296,13 @@ struct MatchTypeReturn {
 // and a r.reason() that describes why it could not match.
 // note: It is possible to successfully match a formal, but for type variables
 // in the formal to still not be defined. In particular, None matches Optional[T]
-// but does not define the value of T. 
+// but does not define the value of T.
 CAFFE2_API MatchTypeReturn
-matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env);
+matchTypeVariables(TypePtr formal, TypePtr actual, TypeEnv& type_env, bool allow_conversions);
```

Comment:
I don't think we need this bool because we still check subtyping relationships after matching type variables. Even if we always allow potential matches, it will still have the same behavior.